x-pack/plugins/rule_registry/server/utils/create_lifecycle_rule_type_factory.ts

```diff
@@ -72,6 +73,12 @@ export const createLifecycleRuleTypeFactory: CreateLifecycleRuleTypeFactory = ({
       } = options;
 
       const ruleExecutorData = getRuleExecutorData(type, options);
+      const so = await options.services.savedObjectsClient.get(
```

Comment:
With [this PR](https://github.com/elastic/kibana/pull/99819), you should be able to access the `rule` object from the executor params and get the `consumer` from there instead of querying the saved objects client directly